### DISCUSSION TITLE
Consider adding an `ignore_tip` or similar config option for forked chains

### DISCUSSION THREAD

**Discussion by oxarbitrage** [ZcashFoundation/zebra#7986] [Ideas]
When forking the zcash testnet chain there is a point in the process where you need to comment out code that checks the chain is close to the tip such as:

https://github.com/ZcashFoundation/zebra/blob/v1.4.0/zebra-rpc/src/methods/get_block_template_rpcs/get_block_template.rs#L179-L199

The Qedit team reported this but i had to do the same when creating a testnet fork. I was wondering if it will worth to add a debug `ignore_tip` config option that could ignore that branch (and potentially others) when activated.

**Comment by teor2345** [comment]
I think that's a good idea. And it should say that Zebra is near the tip regardless of the internal service that's asking.

It could work the same way as the mempool option, which is `debug_enable_at_height`:
https://github.com/ZcashFoundation/zebra/blob/382fb3775addae966e519f95cc82b168e7242bff/zebrad/src/components/mempool.rs#L279

So for example in the QEDIT early testnet fork, they would set it to 20,000, or whatever the block before their fork height is.

**Comment by teor2345** [comment]
As a workaround for now, QEDIT could sync the full testnet chain, and then their fork would be near the tip.

**Reply by arya2** [reply]
They could reset the system clock, get the future blocks and submit them without peers, or start Zebra with peers to avoid commenting out that line, or fork the chain after the max checkpoint height to avoid deleting any checkpoints.

**Comment by oxarbitrage** [comment]
> As a workaround for now, QEDIT could sync the full testnet chain, and then their fork would be near the tip.

It will be out of sync at some point anyways specially if you are experimenting, the workaround for now has been commenting out the code.

**Reply by arya2** [reply]
> It will be out of sync at some point anyways especially if you are experimenting

Yeah it would be difficult to use `debug_stop_at_height` near the tip since the closest the cached state could be is depth 100 and the lowest acceptable depth for the `getblocktemplate` RPC is 101.

They could reset the system clock, get the future blocks and submit them without peers, or start Zebra with peers, but this workaround seems simpler. (Though they could still get around deleting the checkpoints by syncing past the max checkpoint)

**Comment by mpguerra** [comment]
Any outcome yet on this discussion? Should we create an issue from it?

**Comment by oxarbitrage** [comment]
I think this is kind of resolved with regtest/custom testnet support or by some other way as it had been a while and nobody requested this feature anymore.
